Common/src/matrix_structure.cpp

```diff
@@ -732,11 +734,52 @@ void CSysMatrix::ProdBlockVector(unsigned long block_i, unsigned long block_j, c
   
   su2double *block = GetBlock(block_i, block_j);
   
+  switch (nVar) 
```

Comment:
Loops introduce some overhead which can not be compensated for if the loops are short. The compiler has no chance to optimize register usage, introduces a condition each iteration to test for the iteration count. All not important if the loop is long.